Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -1588,6 +1588,14 @@ bool LocalDOMWindow::hasStickyActivation() const
     return now >= m_lastActivationTimestamp;
 }
 
+// When the last history-action activation timestamp of W is not equal to the last activation timestamp of W,
+// then W is said to have history-action activation.
+// (https://html.spec.whatwg.org/multipage/interaction.html#history-action-activation)
+bool LocalDOMWindow::hasHistoryActionActivation() const
```

Comment:
These two functions are currently only reachable by test code.  Their references to the HTML spec indicates that they should actually be hooked up to user interaction code or something like that.

Reply:
It will be used in follow up patches. The close watcher API as well as the navigation API relies on this.